controllers/perconaservermysqlrestore_controller.go

```diff
@@ -113,20 +122,34 @@ func (r *PerconaServerMySQLRestoreReconciler) Reconcile(ctx context.Context, req
 		backup := &apiv1alpha1.PerconaServerMySQLBackup{}
 		nn := types.NamespacedName{Name: cr.Spec.BackupName, Namespace: cr.Namespace}
 		if err := r.Client.Get(ctx, nn, backup); err != nil {
+			if k8serrors.IsNotFound(err) {
+				status.State = apiv1alpha1.RestoreError
+				status.StateDesc = fmt.Sprintf("PerconaServerMySQLBackup %s in namespace %s is not found", cr.Spec.BackupName, cr.Namespace)
+				return ctrl.Result{}, nil
+			}
 			return ctrl.Result{}, errors.Wrapf(err, "get backup %s", nn)
 		}
 		destination = backup.Status.Destination
 		storageName := backup.Spec.StorageName
 		var ok bool
 		storage, ok = cluster.Spec.Backup.Storages[storageName]
 		if !ok {
-			return ctrl.Result{}, errors.Errorf("%s not found in spec.backup.storages in PerconaServerMySQL CustomResource", storageName)
+			status.State = apiv1alpha1.RestoreError
+			status.StateDesc = fmt.Sprintf("%s not found in spec.backup.storages in PerconaServerMySQL CustomResource", storageName)
```

Comment:
Maybe to have this msg like:
`spec.backup.storages does not contain %s`

It will be more inline with the message you have for empty `backupSource.storage.destination` and other messages.